core/src/main/java/org/infinispan/util/logging/Log.java

```diff
@@ -1804,4 +1804,7 @@ CacheConfigurationException offHeapMemoryEvictionSizeNotLargeEnoughForAddresses(
 
    @Message(value = "Maximum startup attempts exceeded for store %s", id = 527)
    PersistenceException storeStartupAttemptsExceeded(String storeName, @Cause Throwable t);
+
+   @Message(value = "Cannot FORCE_WRITE_LOCK as this partition is DEGRADED", id = 528)
```

Comment:
I wonder if we have a similar check for regular locks, maybe we need a more generic message that can be used there as well?

Reply:
Not sure I follow. Where would/does this check exist?

Reply:
It doesn't exist now, but it would be in `PartitionHandlingInterceptor.visitLockControlCommand`. I'm not sure how unavailable owners for lock commands are handled now, but since there's no handling in `PHI` I have a feeling they block waiting for a new topology in `StateTransferInterceptor`.

This got me thinking about `entrySet`/`keySet` as well. They're not really "bulk read" operations, because they return a view over the cache, and only iteration or stream operations fetch all the keys. A long time could pass between the call to keySet/entrySet and their iteration, so I don't think they should perform availability checks. They also ignore the `FORCE_WRITE_LOCK` flag ;)